src/main/scala/ai/privado/languageEngine/go/passes/orm/BaseORMParser.scala

```diff
@@ -51,37 +52,37 @@ abstract class BaseORMParser(cpg: Cpg) extends PrivadoParallelCpgPass[TypeDecl](
     }
   }
 
-  private def buildAndAddSqlQueryNodes(
-    builder: DiffGraphBuilder,
-    model: TypeDecl,
-    fileNode: NodeOrDetachedNode
-  ): Unit = {
+  def buildAndAddSqlQueryNodes(builder: DiffGraphBuilder, model: AstNode, fileNode: NodeOrDetachedNode): Unit = {
+    val typeDeclNode = model.asInstanceOf[TypeDecl]
```

Comment:
Taking AstNode as argument and typeCasting it to TypeDeclNode in base class isn't correct. Instead this function can be made abstract and the class extending this can use it as per there use case.
`def buildAndAddSqlQueryNodes(builder: DiffGraphBuilder, node: AstNode, fileNode: NodeOrDetachedNode): Unit = ???`

Reply:
- `TypeDecl` object is required when we are processing any ORM framework.
- `AstNode` object is required when we are processing plain SQL queries(SQLQueryParser).
-  Apart from `buildAndAddSqlQueryNodes` method every other method has complete different implementation.
-  I think we should inherit `BaseORMParser` only in cases of ORM framework processing.
- We should also think of creating parent class for processing plain SQL queries which can be inherited by other languages too.

In Commit: 2d7cb9d5f36f912f5b0a2a71d5b9ea9e233cb263 I have removed inheritance of BaseORMParser from `SQLQueryParser`.